### DISCUSSION TITLE
How to get access to merged output result of Nuxt related files in the project that uses extends?

### DISCUSSION THREAD

**Discussion by AndrewBogdanovTSS** [nuxt/nuxt#16215] [Questions] [answered]
Hi guys, I will try to describe the scenario I want to achieve and maybe you could recommend how to best do it or whether some new feature should be implemented, so:
**The goal:** Compile SVG sprite file from the public/icons directory
**Project structure:** 
```
base/public/icons
app/public/icons
```
Lib for generating the sprite: https://www.npmjs.com/package/svg-sprite

nuxt.config.ts
```
export default defineNuxtConfig({
    extends: ['.app/', './base']
})
```

Using **extends** I assume that Nuxt creates a resulting **public** directory that would have contents from both folders, but how can I access that content from a point on another Node process? What I would like is to wait till Nuxt will complete a merge action and then point my script to the newly created folder. I want to have such behavior in both **dev** and **build**

**Comment by pi0** [comment, marked as answer]
Hi. During development, we don't copy over assets but serve them separately ([src](https://github.com/unjs/nitro/blob/b332bd37f1dc17cd8386208f8a104b15bea6f5ff/src/dev/server.ts#LL117C3-L117C3)) also production copy happens after nuxt build which might be too late for a nuxt module.

If you need a consistent behavior for your module in dev and production and also merged results **during or before nuxt build**, i think the best would be implementing [copyPublicAssets](https://github.com/unjs/nitro/blob/b332bd37f1dc17cd8386208f8a104b15bea6f5ff/src/build.ts#L34) in your module using `useNitro()`. `nitro.options.publicAssets` contains an array of all layers.
